core/model/upgradeseries.go

```diff
@@ -7,6 +7,83 @@ import (
 	"github.com/juju/errors"
 )
 
+// UpgradeSeriesStatus is a status that a machine or unit can be in during the
+// execution of an OS series upgrade on its host machine.
+type UpgradeSeriesStatus int
+
+// String returns an indicative description of the status.
+func (s UpgradeSeriesStatus) String() string {
+	return upgradeSeriesStatusNames[s]
+}
+
+// Enumerating the status values allows us to represent relative progression
+// though the upgrade-series workflow.
+const (
+	UpgradeSeriesNotStarted UpgradeSeriesStatus = iota
+	UpgradeSeriesPrepareStarted
+	UpgradeSeriesPrepareMachine
+	UpgradeSeriesPrepareComplete
+	UpgradeSeriesCompleteStarted
+	UpgradeSeriesComplete
+	UpgradeSeriesError
+)
```

Comment:
This would require extra machinery to get human-readable data serialized to the database. While I do agree that enumerating the constants would statically indicate the ordering of the states it may not be strictly necessary and require the aforementioned extra marshaling work. Since there is a strong desire by @howbazaar and others to serialize to human-readable formats - It would probably be best to simply leave them as string constants. What are your thoughts on this?

Reply:
From my email:
https://play.golang.org/p/KqKgKUk_syr

This is a common pattern, and I don't think there is any logging concern.

Reply:
I'm not a big fan of the "iota" pattern, because it takes away the fact that the values actually have meaning, and its *really important* that they not change. (adding a value in the middle of the list will continue to pass the test suite, but completely break upgrades/interop with other versions.)

Given we already have a list of strings associated with the constants, why not just use them instead?

Reply:
I've changed it back to a string.